smartsim/_core/_cli/build.py

```diff
@@ -352,31 +361,43 @@ def check_py_torch_version(
             "torchvision": f"{self.versions.TORCHVISION}{device_suffix}",
         }
 
-        def torch_validator(package: str, version: t.Optional[str]) -> bool:
+        self._check_py_package_version(
+            torch_packages,
+            end_point=end_point,
+            validate_installed_version=self.__create_torch_version_validator(
+                with_suffix=device_suffix
+            ),
+            install_on_absent=True,
+            install_on_conflict=handle_conflict,
+        )
+
+    def __create_torch_version_validator(
+        self, with_suffix: str
+    ) -> t.Callable[[str, t.Optional[str]], bool]:
+        def check_torch_version(package: str, version: t.Optional[str]) -> bool:
             if not self.build_env.check_installed(package, version):
                 return False
-            # Previous check only looks at major/minor version numbers,
+            # Default check only looks at major/minor version numbers,
             # Torch requires we look at the patch as well
             installed = self.build_env.get_package_version(package)
-            if device_suffix and device_suffix not in installed.patch:
+            if with_suffix and with_suffix not in installed.patch:
                 # Torch requires that we compare to the patch
                 raise VersionConflictError(
-                    f"{package}=={installed} does not satisfy device "
-                    f"suffix requirement: {device_suffix}"
+                    package,
+                    installed,
+                    version or f"X.X.X{with_suffix}",
+                    msg=(
+                        f"{package}=={installed} does not satisfy device "
```

Comment:
Just a note that the `==` here will eventually have to be tweaked

Reply:
Also, can we get this message down with the _check_py_package_version() method somehow?